mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ExecutorStateManagerImpl.java

```diff
@@ -80,10 +85,14 @@ public int getTotalWorkers() {
     }
 
     /**
-     * Cache the available executors ready to accept assignments. Note these executors' state are not strongly
-     * synchronized and requires state level check when matching.
+     * Cache the available executors ready to accept assignments.
      */
-    private final SortedMap<Double, NavigableSet<TaskExecutorHolder>> executorByCores = new ConcurrentSkipListMap<>();
+    private final Map<TaskExecutorGroupKey, NavigableSet<TaskExecutorHolder>> executorByGroupKey = new HashMap<>();
+
+    // TODO(fdichiara): make this configurable
+    private final FitnessCalculator fitnessCalculator = new CpuWeightedFitnessCalculator();
+
+    private final Map<String, String> assignmentAttributesAndDefaults;
```

Comment:
I feel like we should be consistent with the naming. It feels like we have three different names for the same concept: scheduling.*, assignment.*, allocation.*.

Reply:
lol. you're right. let me make things a bit more consistent.